diff --git a/core/modules/filter/src/Entity/FilterFormat.php b/core/modules/filter/src/Entity/FilterFormat.php
index 697d944100..e72f72b229 100644
--- a/core/modules/filter/src/Entity/FilterFormat.php
+++ b/core/modules/filter/src/Entity/FilterFormat.php
@@ -140,6 +140,13 @@ public function id() {
    */
   public function filters($instance_id = NULL) {
     if (!isset($this->filterCollection)) {
+      foreach ($this->filters as $filter => $filter_definition) {
+        $provider = $filter_definition['provider'];
+        if ($provider && !in_array($provider, ['core', 'component']) && !\Drupal::service('module_handler')->moduleExists($provider)) {
+          unset($this->filters[$filter]);
+        }
+      }
+
       $this->filterCollection = new FilterPluginCollection(\Drupal::service('plugin.manager.filter'), $this->filters);
       $this->filterCollection->sort();
     }
