diff --git a/src/bootstrap-fast.php b/src/bootstrap-fast.php
index 4967477..fc9823f 100644
--- a/src/bootstrap-fast.php
+++ b/src/bootstrap-fast.php
@@ -20,3 +20,14 @@ $root = $finder->getDrupalRoot();

 // Register more namespaces, as needed.
 # $class_loader->addPsr4('Drupal\Tests\my_module\\', "$root/modules/custom/my_module/tests/src");
+
+// Register namespaces for all custom modules on this project.
+foreach (glob("$root/modules/custom/*", GLOB_ONLYDIR) as $path) {
+  $module = basename($path);
+  $class_loader->addPsr4('Drupal\Tests\\' . $module . '\\', "$root/modules/custom/$module/tests/src");
+}
+// Do the same for core tests, since we might want to use some of their traits.
+foreach (glob("$root/core/modules/*", GLOB_ONLYDIR) as $path) {
+  $module = basename($path);
+  $class_loader->addPsr4('Drupal\Tests\\' . $module . '\\', "$root/core/modules/$module/tests/src");
+}
